[cfg,java] Update to latest go-perun.

ggwpez opened the pull request:
- The 100ms sleep can be removed in the latest version of go-perun.
 - Update API functions

ggwpez committed:
[cfg] Update to go-perun v0.8.0

Signed-off-by: Oliver Tale-Yazdi <oliver@perun.network>

ggwpez committed:
[go,java] Update to go-perun v0.8.0

- Add TxFinalityDepth
- Add BlockTimeMs for configurable block time
- Use the 'Watcher' for setting up a Client
- Remove 100ms sleep before sending channel updates
- Update 'Settle' to not call 'Register' anymore
- Update 'HandleUpdate' callback
- Update usage of 'Funder' interface

Signed-off-by: Oliver Tale-Yazdi <oliver@perun.network>

cryptphil commented on android/app/src/androidTest/java/PrnmTest.java:
```diff
@@ -146,7 +146,7 @@ public void sharedSetup(Setup s) throws Exception {
 
         String ethUrl = "ws://10.5.0.9:8545";
-        Config cfg = new Config(Setup.Aliases[s.Index], onChain, s.Adjudicator, s.Assetholder, ethUrl, "0.0.0.0", 5750);
+        Config cfg = new Config(Setup.Aliases[s.Index], onChain, s.Adjudicator, s.Assetholder, ethUrl, "0.0.0.0", 5750, s.TxFinalityDepth);
```
Access `TxFinalityDepth` by the class `Setup` instead of an instance reference since this member is static.
```suggestion
        Config cfg = new Config(Setup.Aliases[s.Index], onChain, s.Adjudicator, s.Assetholder, ethUrl, "0.0.0.0", 5750, Setup.TxFinalityDepth);
```

cryptphil commented on android/app/src/androidTest/java/PrnmTest.java:
```diff
@@ -160,11 +160,10 @@ public void sharedSetup(Setup s) throws Exception {
 
     public void proposeChannel(Setup s) throws Exception {
-        Thread.sleep(5000);
+        Thread.sleep(5000 + s.BlockTimeMs * s.TxFinalityDepth);
```
```suggestion
        Thread.sleep(5000 + Setup.BlockTimeMs * Setup.TxFinalityDepth);
```

cryptphil commented on android/app/src/androidTest/java/PrnmTest.java:
```diff
@@ -191,11 +190,11 @@ public void sendTx(boolean wait, int amount) throws Exception {
             ch.get().finalize(ctx);
         } else {
-            Thread.sleep(5000);
+            Thread.sleep(5000 + s.BlockTimeMs * s.TxFinalityDepth);
```
```suggestion
            Thread.sleep(5000 + Setup.BlockTimeMs * Setup.TxFinalityDepth);
```

cryptphil commented on android/app/src/androidTest/java/PrnmTest.java:
```diff
@@ -191,11 +190,11 @@ public void sendTx(boolean wait, int amount) throws Exception {
     public void sharedTeardown(Setup s) throws Exception {
         if (s.Index == 1) {
-            Thread.sleep(1000);
+            Thread.sleep(1000 + s.BlockTimeMs * s.TxFinalityDepth);
```
```suggestion
            Thread.sleep(1000 + Setup.BlockTimeMs * Setup.TxFinalityDepth);
```

cryptphil commented on channel_update.go:
```diff
@@ -26,8 +29,8 @@ type (
 	ChannelUpdate struct {
-		State    *State // Proposed new state.
-		ActorIdx int    // Who is transferring funds.
+		Last, State *State // Proposed new state.
```
Update the comment:
```suggestion
		Last, State *State // The last state and the proposed new state.
```

cryptphil commented on android/app/src/main/java/network/perun/app/MainActivity.java:
```diff
@@ -52,8 +52,10 @@ protected void onCreate(Bundle savedInstanceState) {
             // in this case we already deployed them and enter their addresses.
             Address adjudicator = new Address("0xDc4A7e107aD6dBDA1870df34d70B51796BBd1335");
             Address assetholder = new Address("0xb051EAD0C6CC2f568166F8fEC4f07511B88678bA");
```
Maybe camelcase for the asset holder?
```suggestion
            Address assetHolder = new Address("0xb051EAD0C6CC2f568166F8fEC4f07511B88678bA");
```

cryptphil reviewed (changes requested):
LGTM, mostly just some style comments. 🚀

ggwpez commented:
I addressed all comments, changes are in the last [force push](https://github.com/perun-network/perun-eth-mobile/compare/9af62680b7389ac4d84db4d503d2b8f15c6afe6a..470203dd869f94be5e8fab5a4d15915519ccab63).  
@cryptphil

cryptphil reviewed (approved):
Perfect! 🎉
